control/autoware_joy_controller/src/autoware_joy_controller/autoware_joy_controller_node.cpp

```diff
@@ -296,124 +300,139 @@ void AutowareJoyControllerNode::publishGateMode()
-  vehicle_cmd.header.stamp = ros::Time::now();
+  autoware_vehicle_msgs::msg::VehicleCommand vehicle_cmd;
+  vehicle_cmd.header.stamp = rclcpp::Clock().now();
   vehicle_cmd.control = prev_control_command_;
   vehicle_cmd.shift.data = prev_shift_;
 
-  pub_vehicle_command_.publish(vehicle_cmd);
+  pub_vehicle_command_->publish(vehicle_cmd);
 }
 
 void AutowareJoyControllerNode::publishRawVehicleCommand()
 {
-  autoware_vehicle_msgs::RawVehicleCommand vehicle_cmd;
-  vehicle_cmd.header.stamp = ros::Time::now();
+  autoware_vehicle_msgs::msg::RawVehicleCommand vehicle_cmd;
+  vehicle_cmd.header.stamp = rclcpp::Clock().now();
   vehicle_cmd.control = prev_raw_control_command_;
   vehicle_cmd.shift.data = prev_shift_;
 
-  pub_raw_vehicle_command_.publish(vehicle_cmd);
+  pub_raw_vehicle_command_->publish(vehicle_cmd);
+}
+
+void AutowareJoyControllerNode::initTimer(double period_s)
+{
+  auto timer_callback = std::bind(&AutowareJoyControllerNode::onTimer, this);
+  const auto period_ns =
+    std::chrono::duration_cast<std::chrono::nanoseconds>(std::chrono::duration<double>(period_s));
+  timer_ = std::make_shared<rclcpp::GenericTimer<decltype(timer_callback)>>(
+    this->get_clock(), period_ns, std::move(timer_callback),
+    this->get_node_base_interface()->get_context());
+  this->get_node_timers_interface()->add_timer(timer_, nullptr);
 }
 
 AutowareJoyControllerNode::AutowareJoyControllerNode()
+: Node("autoware_joy_controller")
 {
   // Parameter
-  private_nh_.param<std::string>("joy_type", joy_type_, "DS4");
-  private_nh_.param("update_rate", update_rate_, 10.0);
-  private_nh_.param("accel_ratio", accel_ratio_, 3.0);
-  private_nh_.param("brake_ratio", brake_ratio_, 5.0);
-  private_nh_.param("steer_ratio", steer_ratio_, 0.5);
-  private_nh_.param("steering_angle_velocity", steering_angle_velocity_, 0.1);
-  private_nh_.param("control_command/velocity_gain", velocity_gain_, 3.0);
-  private_nh_.param("control_command/max_forward_velocity", max_forward_velocity_, 20.0);
-  private_nh_.param("control_command/max_backward_velocity", max_backward_velocity_, 3.0);
-  private_nh_.param("control_command/backward_accel_ratio", backward_accel_ratio_, 1.0);
-
-  ROS_INFO("Joy type: %s", joy_type_.c_str());
+  joy_type_ = declare_parameter("joy_type", std::string("DS4"));
+  update_rate_ = declare_parameter("update_rate", 10.0);
+  accel_ratio_ = declare_parameter("accel_ratio", 3.0);
+  brake_ratio_ = declare_parameter("brake_ratio", 5.0);
+  steer_ratio_ = declare_parameter("steer_ratio", 0.5);
+  steering_angle_velocity_ = declare_parameter("steering_angle_velocity", 0.1);
+  velocity_gain_ = declare_parameter("control_command/velocity_gain", 3.0);
+  max_forward_velocity_ = declare_parameter("control_command/max_forward_velocity", 20.0);
+  max_backward_velocity_ = declare_parameter("control_command/max_backward_velocity", 3.0);
+  backward_accel_ratio_ = declare_parameter("control_command/backward_accel_ratio", 1.0);
```

Comment:
I think you can move this to the initializer list.

Reply:
as with other nodes in the code base, for consistency in all the code I've left them in the constructor body